### PR TITLE
Add Roles and Permissions endpoints

### PR DESCRIPTION
**Pull request by lbalmaceda**
### Changes

Adds all missing endpoints for **Roles** and User/Role **Permissions**.

### Testing

- [x] This change adds unit test coverage
- [x] This change adds integration test coverage
- [x] This change has been tested on the latest version of the platform/language or why not

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [x] All existing and new tests complete without errors

### CODE REVIEW

File: auth0/v3/management/users.py [hunk 1]
```diff
@@ -126,18 +129,31 @@ def update(self, id, body):
         """
         return self.client.patch(self._url(id), data=body)
 
-    def get_roles(self, id):
+    def list_roles(self, id, page=0, per_page=25, include_totals=True):
```
Comment (lbalmaceda) [inline auth0/v3/management/users.py]: "roles" file was not release yet. We can still change the public API

File: auth0/v3/management/users.py [hunk 2]
```diff
@@ -165,6 +181,58 @@ def assign_roles(self, id, roles):
         body = {'roles': roles}
         return self.client.post(url, data=body)
 
+    def list_permissions(self, id, page=0, per_page=25, include_totals=True):
+        """Get a user's permissions.
```
Comment (damieng) [inline auth0/v3/management/users.py]: Maybe put `List a user's permissions` ?

File: auth0/v3/management/users.py [hunk 3]
```diff
@@ -165,6 +181,58 @@ def assign_roles(self, id, roles):
...
+        body = {'permissions': permissions}
+        return self.client.delete(url, data=body)
+
+    def add_permissions(self, id, permissions):
+        """Assign permissions to a user
+
+        Args:
+            id (str): The user's id.
+
+            permissions (list of str): A list of permissions ids to associated with the user.
```
Comment (damieng) [inline auth0/v3/management/users.py]: "permission ids"

File: auth0/v3/management/users.py [hunk 4]
```diff
@@ -165,6 +181,58 @@ def assign_roles(self, id, roles):
...
+        url = self._url('{}/permissions'.format(id))
+        return self.client.get(url, params=params)
+
+    def remove_permissions(self, id, permissions):
+        """Removes permissions from a user.
+
+        Args:
+            id (str): The user's id.
+
+            permissions (list of str): A list of permissions ids to unassociate from the user.
```
Comment (damieng) [inline auth0/v3/management/users.py]: "Permission ids"

File: auth0/v3/management/roles.py [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+        See https://auth0.com/docs/api/management/v2#!/Roles/delete_role_permission_assignment
+        """
+        url = self._url('{}/permissions'.format(id))
+        body = {'permissions': permissions}
+        return self.client.delete(url, data=body)
+
+
+    def add_permissions(self, id, permissions):
+        """Adds permissions from a role.
```
[the remark this answers is not in the document]
Comment (lbalmaceda) [reply auth0/v3/management/roles.py]: Changed to what the API Explorer uses:
>"Associates permissions with a role."

File: auth0/v3/management/users.py (hunk 2 above)
Comment (lbalmaceda) [reply auth0/v3/management/users.py]: Changed to:
> Get the permissions associated to the user.

File: auth0/v3/management/users.py (hunk 3 above)
Comment (lbalmaceda) [reply auth0/v3/management/users.py]: Resolved in 3 other places as well

**Comment by lbalmaceda** [comment]
@damieng will need a force merge

**Pull request merged**